mlflow/data.py

```diff
@@ -65,6 +65,11 @@ def is_uri(string):
     return len(parsed_uri.scheme) > 0
 
 
+def is_action(action_string):
+    pattern = re.compile("^-{1,2}[a-zA-z]+")
```

Comment:
This doesn't cover all the valid Python identifiers (e.g. --action_, --act10n, etc.). I would recommend checking [this SO post](https://stackoverflow.com/questions/5474008/regular-expression-to-confirm-whether-a-string-is-a-valid-identifier-in-python) for a more thorough check.

Reply:
Ideally, we would be able to cover the case where we have an action that is toggled by an argument being present or not (e.g. passing in `--earlystopping` as an argument if we want early stopping, and not passing it in if we don't want early stopping). For example, if I wanted to specify that I don't want early stopping, I would call 
```
mlflow run . -P data_file=data.csv -P stop=''
```

 and if I wanted early stopping I would call
 
```
mlflow run . -P data_file=data.csv -P stop=--earlystopping
```

It'd be good if you could add handling of an indicator (something like an empty string) that signals that the user does not want to pass in the action to the command.

Reply:
An alternative to this implementation (instead of having to deal with actions) I can think of is creating a new `type` (maybe called `flag`?) that takes in `True` or `False` (or equivalent values), and appends the flag to the command run, depending on the value of the `flag`.

For example, with an example MLProject of
```
...
parameters:
      data_file: path
      regularization: {type: float, default: 0.1}
      train: {type: flag, default: False}
    command: "python script.py -r  {regularization} {train} {data_file}"
...
```
a call of `mlflow run . data_file=data.csv` would translate to `python script.py --train -r 0.1 data.csv`, just like your proposal from #1945 proposes.

Reply:
Good point, addressed this in the new commit. Passing an empty string will allow you to toggle the argument